modules/drivers/mcp230/mcp230.js

```diff
@@ -32,138 +32,104 @@ const PULLUPS_8  = 0b00000000;
       // User specific state initialization settings
-      inputs = offset ? INPUTS_16 : INPUTS_8,
-      pullups = offset ? PULLUPS_16 : PULLUPS_8,
-      pins,
-      reg,
-      reg: { IODIR, GPIO, GPPU }
-    } = dictionary || {};
-
-    super({ address, hz, sda, scl });
-
-    // trace(`IODIR: 0x${IODIR.toString(16)}\n`);
-    // trace(`GPIO: 0x${GPIO.toString(16)}\n`);
-    // trace(`GPPU: 0x${GPPU.toString(16)}\n`);
+      inputs = this.offset ? INPUTS_16 : INPUTS_8,
+      pullups = this.offset ? PULLUPS_16 : PULLUPS_8,
+    } = dictionary;
 
     // If the value of inputs does not match the default,
     // then set the user defined inputs configuration
-    if ((inputs !== (offset ? INPUTS_16 : INPUTS_8))) {
-      this.writeByte(IODIR, inputs);
+    if ((inputs !== (this.offset ? INPUTS_16 : INPUTS_8))) {
+      this.writeByte(this.IODIR, inputs);
 
-      if (offset) {
-        this.writeByte(IODIR + offset, inputs >> 8);
+      if (this.offset) {
+        this.writeByte(this.IODIR + this.offset, inputs >> 8);
       }
     }
 
     // If the value of pullups does not match the default,
     // then set the user defined pullups configuration
-    if ((pullups !== (offset ? PULLUPS_16 : PULLUPS_8))) {
-      this.writeByte(GPPU, pullups);
+    if ((pullups !== (this.offset ? PULLUPS_16 : PULLUPS_8))) {
+      this.writeByte(this.GPPU, pullups);
 
-      if (offset) {
-        this.writeByte(GPPU + offset, pullups >> 8);
+      if (this.offset) {
+        this.writeByte(this.GPPU + this.offset, pullups >> 8);
       }
     }
 
-    for (let pin = 0; pin < pins; pin++) {
+    for (let pin = 0; pin < this.length; pin++) {
       this[pin] = new Pin({ pin, expander: this });
     }
 
-    this.length = pins;
-    this.offset = offset;
-    this.reg = Object.freeze(reg);
-
     Object.freeze(this);
   }
 
-  * [Symbol.iterator]() {
-    let i = 0;
-    while (i < this.length) {
-      yield this[i];
-      i++;
-    }
-  }
-
   bankWrite(state) {
-    const { IODIR, GPIO } = this.reg;
-
     if (this.offset) {
       // Read IODIR state
-      let iodir = this.readWord(IODIR);
+      let iodir = this.readWord(this.IODIR);
 
       // Set IODIR state to OUTPUT
-      this.writeWord(IODIR, 0x0000);
+      this.writeWord(this.IODIR, 0x0000);
 
       // Write GPIO
-      this.writeWord(GPIO, state);
+      this.writeWord(this.GPIO, state);
 
       // Restore previous IODIR state
-      this.writeWord(IODIR, iodir);
+      this.writeWord(this.IODIR, iodir);
     } else {
       // Read IODIR state
-      let iodir = this.readByte(IODIR);
+      let iodir = this.readByte(this.IODIR);
 
       // Set IODIR state to OUTPUT
-      this.writeByte(IODIR, 0x00);
+      this.writeByte(this.IODIR, 0x00);
 
       // Write GPIO
-      this.writeByte(GPIO, state & 0xFF);
+      this.writeByte(this.GPIO, state & 0xFF);
 
       // Restore previous IODIR state
-      this.writeByte(IODIR, iodir);
+      this.writeByte(this.IODIR, iodir);
     }
```

Comment:
Updates per **Device specific constants** section of https://github.com/Moddable-OpenSource/moddable/pull/21#issuecomment-353265714